### ISSUE TITLE
Return data ?

### ISSUE THREAD

**Issue by tony-go** [react-spring/react-nil#5]
This is more a question than an issue. (sorry for that)

Are those "components" should not return data? Or should `react-nil` provide a new type of "components" that return data (object, array,...) instead of return null.

Just an idea.

**Comment by drcmda** [comment]
in react components do not return data but v-dom which goes into the hands of the reconciler. there is no component that receives the return value of another. 

you have props

```jsx
function Foo({ data }) {
  return <Bar data={...} />
```

render props

```jsx
<Foo>
  {data => ( ... )}
</Foo>
```

- context
- ref + useImperativeHandle
- your state manager of choice

i have seen a mini react for node whose components do return data, the problem is that A. this seems kinda shaky because components are async, and B. it would sacrifice the entire react eco system. it could not just go and use redux, react-graphql or whatever on the server.

**Comment by tony-go** [comment]
Props and render props could work in fact ^^

Thanks for the response ^^

**Comment by CodyJasonBennett** [comment]
Implemented this in #12.
